### ISSUE TITLE
Improve the 1, 2, 3 in the README

### ISSUE THREAD

**Issue by rvolgers** [salsa-rs/salsa#187]
Part 3 of the 1, 2, 3 is confusing to me, as it suddenly talks about a "query struct" (which has not been previously mentioned) when talking about the database.

https://github.com/salsa-rs/salsa/blob/57174e2c63d8aeb0dc95ddb0647533b8274119b4/README.md#L44-L47

My "reading the comments in the helloworld example" skills tell me that it may be referring to storage being split out in structs per query group, and the database just puts all those together, but I actually know nothing about Salsa so it's very possible that I'm wrong.

TLDR: I can count to three, but I can't understand this 1, 2, 3.

**Comment by rvolgers** [comment]
(Updated issue title so it rhymes.)

**Comment by Veykril** [comment]
This is no longer relevant
